### ISSUE TITLE
Getting compilation error in latest version but works fine in v1.51.0

### ISSUE THREAD

**Issue by shahincsejnu** [golangci/golangci-lint#3573]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

I'm getting some compilation error in latest version from GH action, errors are like:
```
field `X` is unused (unused)
filed `Y` is unused (unused)
```
even though they got used, basically in the previous version (like: `v1.51.0`) it's working fine but when tried to use latest(`v1.51.1`) it's failing there!

the config of my GH site looks like:
```
      - name: golangci-lint
        uses: golangci/golangci-lint-action@v2
        with:
          version: latest
          working-directory: auth/
          skip-go-installation: true
          skip-pkg-cache: true
```

I'm using Go version `1.18`

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
# Paste output here
basically it's in GH action and latest(`v1.51.1`) one as I attached the GH action part in above
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
# paste output here
```

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
1.18
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```

</details>

### Code example or link to a public repository

<details>

```go
// add your code here
```

</details>

**Comment by ldez** [comment]
duplicate of #3569
